Auto update failing for tree-sitter-lua

Hi, I'm Termux 🤖.

I'm here to help you update your Termux packages.

I've tried to update the [tree-sitter-lua](https://github.com/termux/termux-packages/blob/29e4cf9d2f23d45f57a13e01ad68d9c99f387f17/packages/tree-sitter-lua) package, but it failed.

Here's the output of the update script:
<details>
<summary>Show log</summary>
<pre lang="console">
INFO: Updating tree-sitter-lua [Current version: 0.3.0]
INFO: package being updated to 0.4.0.
Downloading https://github.com/tree-sitter-grammars/tree-sitter-lua/archive/refs/tags/v0.4.0.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

 19 63999   19 12356    0     0  57784      0  0:00:01 --:--:--  0:00:01 57784
100 63999  100 63999    0     0   254k      0 --:--:-- --:--:-- --:--:-- 1626k

You are about to commit these changes:

--------------------
diff --git a/packages/tree-sitter-lua/build.sh b/packages/tree-sitter-lua/build.sh
index 6ef620b367..f1615b659b 100644
--- a/packages/tree-sitter-lua/build.sh
+++ b/packages/tree-sitter-lua/build.sh
@@ -2,10 +2,9 @@ TERMUX_PKG_HOMEPAGE=https://github.com/tree-sitter-grammars/tree-sitter-lua
 TERMUX_PKG_DESCRIPTION="Lua grammar for tree-sitter"
 TERMUX_PKG_LICENSE="MIT"
 TERMUX_PKG_MAINTAINER="Joshua Kahn @TomJo2000"
-TERMUX_PKG_VERSION="0.3.0"
-TERMUX_PKG_REVISION=1
+TERMUX_PKG_VERSION="0.4.0"
 TERMUX_PKG_SRCURL=https://github.com/tree-sitter-grammars/tree-sitter-lua/archive/refs/tags/v${TERMUX_PKG_VERSION}.tar.gz
-TERMUX_PKG_SHA256=a34cc70abfd8d2d4b0fabf01403ea05f848e1a4bc37d8a4bfea7164657b35d31
+TERMUX_PKG_SHA256=b0977aced4a63bb75f26725787e047b8f5f4a092712c840ea7070765d4049559
 TERMUX_PKG_AUTO_UPDATE=true
 TERMUX_PKG_UPDATE_TAG_TYPE='newest-tag'
 TERMUX_PKG_BUILD_IN_SRC=true
--------------------

bump(main/tree-sitter-lua): 0.4.0


Not committing to Git!

INFO: Trying to build package.
INFO: Package tree-sitter-lua exists in main repo.
Running container 'termux-package-builder' from image 'ghcr.io/termux/package-builder'...
termux - building tree-sitter-lua for arch aarch64...
::group::INFO: Fetching repo metadata
[packages-cf.termux.dev-apt-termux-main]: Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release
[packages-cf.termux.dev-apt-termux-main]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[packages-cf.termux.dev-apt-termux-main]:                                  Dload  Upload   Total   Spent    Left  Speed
[packages-cf.termux.dev-apt-termux-main]: 
[packages-cf.termux.dev-apt-termux-main]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-main]: 100 13134  100 13134    0     0   208k      0 --:--:-- --:--:-- --:--:--  210k
[packages-cf.termux.dev-apt-termux-main]: Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release.gpg
[packages-cf.termux.dev-apt-termux-main]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[packages-cf.termux.dev-apt-termux-main]:                                  Dload  Upload   Total   Spent    Left  Speed
[packages-cf.termux.dev-apt-termux-main]: 
[packages-cf.termux.dev-apt-termux-main]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-main]: 100   833  100   833    0     0  12881      0 --:--:-- --:--:-- --:--:-- 13015
[packages-cf.termux.dev-apt-termux-main]: gpg: Signature made Sat 17 May 2025 06:36:52 AM UTC
[packages-cf.termux.dev-apt-termux-main]: gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
[packages-cf.termux.dev-apt-termux-main]: gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
[packages-cf.termux.dev-apt-termux-main]: gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
[packages-cf.termux.dev-apt-termux-root]: Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release
[packages-cf.termux.dev-apt-termux-root]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[packages-cf.termux.dev-apt-termux-root]:                                  Dload  Upload   Total   Spent    Left  Speed
[packages-cf.termux.dev-apt-termux-root]: 
[packages-cf.termux.dev-apt-termux-root]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-root]: 100 13319  100 13319    0     0   222k      0 --:--:-- --:--:-- --:--:--  224k
[packages-cf.termux.dev-apt-termux-root]: Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release.gpg
[packages-cf.termux.dev-apt-termux-root]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[packages-cf.termux.dev-apt-termux-root]:                                  Dload  Upload   Total   Spent    Left  Speed
[packages-cf.termux.dev-apt-termux-root]: 
[packages-cf.termux.dev-apt-termux-root]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-root]: 100   833  100   833    0     0  12328      0 --:--:-- --:--:-- --:--:-- 12432
[packages-cf.termux.dev-apt-termux-root]: gpg: Signature made Wed 07 May 2025 12:32:36 PM UTC
[packages-cf.termux.dev-apt-termux-root]: gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
[packages-cf.termux.dev-apt-termux-root]: gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
[packages-cf.termux.dev-apt-termux-root]: gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
[packages-cf.termux.dev-apt-termux-x11]: Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release
[packages-cf.termux.dev-apt-termux-x11]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[packages-cf.termux.dev-apt-termux-x11]:                                  Dload  Upload   Total   Spent    Left  Speed
[packages-cf.termux.dev-apt-termux-x11]: 
[packages-cf.termux.dev-apt-termux-x11]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-x11]: 100 13120  100 13120    0     0   311k      0 --:--:-- --:--:-- --:--:--  312k
[packages-cf.termux.dev-apt-termux-x11]: Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release.gpg
[packages-cf.termux.dev-apt-termux-x11]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[packages-cf.termux.dev-apt-termux-x11]:                                  Dload  Upload   Total   Spent    Left  Speed
[packages-cf.termux.dev-apt-termux-x11]: 
[packages-cf.termux.dev-apt-termux-x11]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-x11]: 100   833  100   833    0     0  13111      0 --:--:-- --:--:-- --:--:-- 13222
[packages-cf.termux.dev-apt-termux-x11]: gpg: Signature made Sat 17 May 2025 06:38:13 AM UTC
[packages-cf.termux.dev-apt-termux-x11]: gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
[packages-cf.termux.dev-apt-termux-x11]: gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
[packages-cf.termux.dev-apt-termux-x11]: gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
::endgroup::
Downloading tree-sitter-lua@0.4.0 source from 'https://github.com/tree-sitter-grammars/tree-sitter-lua/archive/refs/tags/v0.4.0.tar.gz' if necessary...
Downloading https://github.com/tree-sitter-grammars/tree-sitter-lua/archive/refs/tags/v0.4.0.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100 63999  100 63999    0     0   265k      0 --:--:-- --:--:-- --:--:--  265k
-- Android: Targeting API '24' with architecture 'arm64', ABI 'arm64-v8a', and processor 'aarch64'
-- The C compiler identification is Clang 18.0.3
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /home/builder/.termux-build/_cache/android-r27c-api-24-v1/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Configuring done (0.2s)
-- Generating done (0.0s)
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_CXX_FLAGS
    CMAKE_FIND_ROOT_PATH
    CMAKE_FIND_ROOT_PATH_MODE_LIBRARY
    CMAKE_USE_SYSTEM_LIBRARIES
    DOXYGEN_EXECUTABLE


-- Build files have been written to: /home/builder/.termux-build/tree-sitter-lua/src
[1/4] Generating parser.c
FAILED: src/parser.c /home/builder/.termux-build/tree-sitter-lua/src/src/parser.c 
cd /home/builder/.termux-build/tree-sitter-lua/src && /usr/bin/tree-sitter generate src/grammar.json --abi=15
thread 'main' panicked at src/generate/render.rs:1647:9:
This version of Tree-sitter can only generate parsers with ABI version 13 - 14, not 15
stack backtrace:
   0:     0x55d20f72af8b - <unknown>
   1:     0x55d20f76f270 - <unknown>
   2:     0x55d20f71f4bd - <unknown>
   3:     0x55d20f72ad74 - <unknown>
   4:     0x55d20f752e2a - <unknown>
   5:     0x55d20f752acd - <unknown>
   6:     0x55d20f753248 - <unknown>
   7:     0x55d20f72b36e - <unknown>
   8:     0x55d20f72b1a6 - <unknown>
   9:     0x55d20f752f72 - <unknown>
  10:     0x55d20f408cb5 - <unknown>
  11:     0x55d20f548cb3 - <unknown>
  12:     0x55d20f4bab18 - <unknown>
  13:     0x55d20f4b9c45 - <unknown>
  14:     0x55d20f40fd05 - <unknown>
  15:     0x55d20f40b4ea - <unknown>
  16:     0x55d20f422563 - <unknown>
  17:     0x55d20f423ee9 - <unknown>
  18:     0x55d20f752e64 - <unknown>
  19:     0x55d20f738a0b - <unknown>
  20:     0x55d20f423ede - <unknown>
  21:     0x7fd198d351ca - <unknown>
  22:     0x7fd198d3528b - __libc_start_main
  23:     0x55d20f409585 - <unknown>
  24:                0x0 - <unknown>
ninja: build stopped: subcommand failed.
ERROR: failed to build.</pre>
</details>

<hr>
<i>
Above error occured when I last tried to update at 2025-05-17 12:15:06 UTC.<br>
Run ID: <a href="https://github.com/termux/termux-packages/actions/runs/15084952529">15084952529</a><br><br>
<b>Note:</b> Automatic updates will be disabled until this issue is resolved.<br>
</i>

--- a/packages/tree-sitter-lua/build.sh
+++ b/packages/tree-sitter-lua/build.sh
@@ -2,13 +2,18 @@ TERMUX_PKG_HOMEPAGE=https://github.com/tree-sitter-grammars/tree-sitter-lua
 TERMUX_PKG_DESCRIPTION="Lua grammar for tree-sitter"
 TERMUX_PKG_LICENSE="MIT"
 TERMUX_PKG_MAINTAINER="Joshua Kahn @TomJo2000"
-TERMUX_PKG_VERSION="0.3.0"
-TERMUX_PKG_REVISION=1
+TERMUX_PKG_VERSION="0.4.0"
 TERMUX_PKG_SRCURL=https://github.com/tree-sitter-grammars/tree-sitter-lua/archive/refs/tags/v${TERMUX_PKG_VERSION}.tar.gz
-TERMUX_PKG_SHA256=a34cc70abfd8d2d4b0fabf01403ea05f848e1a4bc37d8a4bfea7164657b35d31
+TERMUX_PKG_SHA256=b0977aced4a63bb75f26725787e047b8f5f4a092712c840ea7070765d4049559
 TERMUX_PKG_AUTO_UPDATE=true
 TERMUX_PKG_UPDATE_TAG_TYPE='newest-tag'
 TERMUX_PKG_BUILD_IN_SRC=true
+
+termux_step_pre_configure() {
+	if [[ "${TERMUX_ON_DEVICE_BUILD}" != "true" ]]; then
+		export PATH="$TERMUX_PREFIX"/opt/tree-sitter/cross/bin:$PATH
+	fi
+}
 
 termux_step_post_make_install() {
 	install -d "${TERMUX_PREFIX}"/lib/tree_sitter

--- a/packages/tree-sitter/build.sh
+++ b/packages/tree-sitter/build.sh
@@ -3,7 +3,7 @@ TERMUX_PKG_DESCRIPTION="An incremental parsing system for programming tools"
 TERMUX_PKG_LICENSE="MIT"
 TERMUX_PKG_MAINTAINER="Joshua Kahn @TomJo2000"
 TERMUX_PKG_VERSION="0.25.6"
-TERMUX_PKG_REVISION=1
+TERMUX_PKG_REVISION=2
 TERMUX_PKG_SRCURL=https://github.com/tree-sitter/tree-sitter/archive/refs/tags/v${TERMUX_PKG_VERSION}.tar.gz
 TERMUX_PKG_SHA256=ac6ed919c6d849e8553e246d5cd3fa22661f6c7b6497299264af433f3629957c
 TERMUX_PKG_BREAKS="libtreesitter"
@@ -20,6 +20,14 @@ termux_step_post_get_source() {
 	if [[ "$TERMUX_PKG_VERSION" != "${_SOVERSION}".* ]]; then
 		termux_error_exit "SOVERSION guard check failed."
 	fi
+
+	termux_download \
+	https://github.com/tree-sitter/tree-sitter/releases/download/v${TERMUX_PKG_VERSION}/tree-sitter-linux-x64.gz \
+	"$TERMUX_PKG_CACHEDIR"/tree-sitter-linux-x64.gz \
+	c300ea9f2ca368186ce1308793aaad650c3f6db78225257cbb5be961aeff4038
+	gunzip -k -v "$TERMUX_PKG_CACHEDIR"/tree-sitter-linux-x64.gz
+	mv -v "$TERMUX_PKG_CACHEDIR"/{tree-sitter-linux-x64,tree-sitter}
+	install -Dm700 -t "$TERMUX_PREFIX"/opt/tree-sitter/cross/bin "$TERMUX_PKG_CACHEDIR"/tree-sitter
 }
 
 termux_step_pre_configure() {

--- a/packages/tree-sitter/tree-sitter-cross-tools.subpackage.sh
+++ b/packages/tree-sitter/tree-sitter-cross-tools.subpackage.sh
@@ -1,0 +1,5 @@
+TERMUX_SUBPKG_DESCRIPTION="Tools for cross build on the host (NOT for Termux)"
+TERMUX_SUBPKG_PLATFORM_INDEPENDENT=true
+TERMUX_SUBPKG_INCLUDE="
+opt/tree-sitter/cross
+"

--- a/scripts/setup-ubuntu.sh
+++ b/scripts/setup-ubuntu.sh
@@ -319,9 +319,6 @@ PACKAGES+=" swig"
 # Needed by binutils-cross
 PACKAGES+=" libzstd-dev"
 
-# Needed by tree-sitter-c
-PACKAGES+=" tree-sitter-cli"
-
 # Needed by wlroots
 PACKAGES+=" glslang-tools"
 

